### PR TITLE
Backport triv pr15

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: src/test/denialofservice_tests.cpp [hunk 1]
```diff
@@ -65,7 +65,6 @@ BOOST_AUTO_TEST_CASE(outbound_slow_chain_eviction)
     dummyNode1.SetSendVersion(PROTOCOL_VERSION);
 
     peerLogic->InitializeNode(&dummyNode1);
-    dummyNode1.nVersion = 1;
```
Comment (PastaPastaPasta) [inline src/test/denialofservice_tests.cpp:68]: drop 20131, needs 17785

File: contrib/signet/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Contents
+========
+This directory contains tools related to Signet, both for running a Signet yourself and for using one.
```
Comment (PastaPastaPasta) [inline contrib/signet/README.md:3]: drop 20145, we don't have signnet

File: doc/i2p.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# I2P support in Bitcoin Core
+
+It is possible to run Bitcoin Core as an
```
Comment (PastaPastaPasta) [inline doc/i2p.md:3]: Not sure if I should drop 22250 or dashify, I don't THINK we support i2p at this point, do we?

File: src/test/denialofservice_tests.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/test/denialofservice_tests.cpp:68]: agree

File: src/core_read.cpp [hunk 4]
```diff
@@ -41,8 +40,9 @@ CScript ParseScript(const std::string& s)
             std::string strName(name);
             mapOpNames[strName] = static_cast<opcodetype>(op);
             // Convenience: OP_ADD and just ADD are both recognized:
-            boost::algorithm::replace_first(strName, "OP_", "");
-            mapOpNames[strName] = static_cast<opcodetype>(op);
+            if (strName.compare(0, 3, "OP_") == 0) {  // strName starts with "OP_"
```
Comment (UdjinM6) [inline src/core_read.cpp:43]: 20067: let's leave it to #4246

File: contrib/signet/README.md (hunk 2 above)
Comment (UdjinM6) [reply contrib/signet/README.md:3]: agree

File: src/net_processing.cpp [hunk 5]
```diff
@@ -4467,7 +4467,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
         }
 
         //
-        // Message: getdata (non-blocks)
+        // Message: getdata (transactions)
```
Comment (UdjinM6) [inline src/net_processing.cpp:4470]: 15451: DNM, we use getdata for various Dash specific messages

File: doc/i2p.md (hunk 3 above)
Comment (UdjinM6) [reply doc/i2p.md:3]: we don't, should drop 22250 for now

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
Resolved all comments @UdjinM6 fped

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**